### PR TITLE
build(aio): add initial support for generating guides

### PR DESCRIPTION
**Pull request by petebacondarwin**
This PR demonstrates how we are going to render the guide docs.
It takes a single cookbook article (and its associated examples) and generates HTML from it.

This uses the Rho markdown rendering engine, which has a few bugs. But I have two open PRs to fix those right now: https://github.com/inca/rho/pulls

### CODE REVIEW

File: tools/docs/angular.io-package/index.js [hunk 1]
```diff
@@ -47,13 +48,15 @@ module.exports =
         .config(function(checkAnchorLinksProcessor, log) {
           // TODO: re-enable
           checkAnchorLinksProcessor.$enabled = false;
+          // log.level = 'silly';
```
Comment (petebacondarwin) [inline tools/docs/angular.io-package/index.js]: oops need to remove this.

File: tools/docs/angular.io-package/index.js [hunk 2]
```diff
@@ -47,13 +48,15 @@ module.exports =
...
         })
 
         // Where do we get the source files?
         .config(function(
             readTypeScriptModules, readFilesProcessor, collectExamples, generateKeywordsProcessor) {
 
           // API files are typescript
+          readTypeScriptModules.$enabled = false;
```
Comment (petebacondarwin) [inline tools/docs/angular.io-package/index.js]: oops need to remove this

**Comment by wardbell** [comment]
We talked about this but thought I should leave a note here. Many of these files are sample boilerplate that should not be checked in here.  Pete is going to purify this PR and bring the sample generation-and-maintenance infrastructure over from angular.io in a prudent way.

I'm not sure if this PR yet reflects these intentions as the folder structure may be different here in angular/angular

**Comment by IgorMinar** [comment]
@petebacondarwin CI is failing.

@wardbell I'm fine merging this if necessary for iterating. all this content stuff will be replaced anyway once we merge the repos.

**Review by IgorMinar — APPROVED** [review]
please fix the CI failure and respond to @wardbell comment

**Comment by petebacondarwin** [comment]
@IgorMinar I have addressed the CI problem.
@wardbell I have removed the boilerplate files. It may be that we move the examples to another folder later once your work on the information architecture is complete.

**Pull request merged**